File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        if not root:
            return []
        
        stack = deque([root])
        result = []

        while stack:
            level = []

            for _ in range(len(stack)):
                node = stack.popleft()
                level.append(node.val)

                if node.left:
                    stack.append(node.left)
                if node.right:
                    stack.append(node.right)

            result.append(level)

        return result


        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.cpp
class Solution {
   public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        if (numbers.empty()) {
            return {};
        }

        int left = 0;
        int right = numbers.size() - 1;

        while (left < right) {
            int value = numbers[left] + numbers[right];

            if (value == target) {
                return {left + 1, right + 1};
            }
            if (value < target) {
                left++;
            }
            if (value > target) {
                right--;
            }
        }
    }
};
